include/zephyr/drivers/video-controls.h

```diff
@@ -171,6 +171,9 @@ enum video_exposure_auto_type {
 /** Selection of the type of test pattern to represent */
 #define VIDEO_CID_TEST_PATTERN (VIDEO_CID_IMAGE_PROC_CLASS_BASE + 3)
 
+/** Link frequency, applicable for the CSI2 based devices. This control is read-only. */
+#define VIDEO_CID_LINK_FREQUENCY (VIDEO_CID_IMAGE_PROC_CLASS_BASE + 4)
```

Comment:
In Linux, `V4L2_CID_IMAGE_PROC_CLASS_BASE + 4` is [`V4L2_CID_DEINTERLACING_MODE`](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/include/uapi/linux/v4l2-controls.h?h=v6.15-rc6#n1233).

There is this comment above:
https://github.com/zephyrproject-rtos/zephyr/blob/b5934e8dfe5c8dd4c1a9f2221341aa46e3dce72a/include/zephyr/drivers/video-controls.h#L22-L28

So we can go low-effort mode and import the content unchanged (for the doc also?) :)

```suggestion
/** The frequency of the data bus (e.g. parallel or CSI-2). This control is read-only. */
#define VIDEO_CID_LINK_FREQ (VIDEO_CID_IMAGE_PROC_CLASS_BASE + 4)
```

Reply:
From a licence point of view, is that alright to have stuff from Linux into Zephyr ? I understand that using same numbering or name could help when creating stuff on top of the CTRL API but is that ok ?
Regarding importing the doc as well, I don't think that is acceptable considering that this would really be copy of contents from an incompatible licence.

Reply:
I think the license on the specification is applied for the spec itself, i.e. we need to follow it when making another spec based on it (derivative works). When it comes to implementing a specification in code, I think it is not considered a derivative work of the specification, so the license is not applied. I think this is the purpose of a specification, i.e. everyone needs to follow it to make the implementation (e.g. each vendor implements the MPEG spec differently) ... how we can follow a spec without using it (?).

However, I think we need other experts to explain these things and advise how to do now, especially with the documentation.

Reply:
To split the PR and longer-term discussion, would it be a suitable compromise to keep the Linux CIDs but leave the documentation empty until this is resolved?

Reply:
Nice ! For the copyright, we need to keep the original copyright ?